doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,375 @@
+offloading operations to the system kernel whenever possible.  
+
+Since most modern kernels are multi-threaded, they can handle
+multiple operations executing in the background.  When one of these
+operations completes, the kernel tells Node.js so that the appropriate callback 
+may added to the `poll` queue to eventually be executed.  We'll explain
+this in further detail later in this topic.  
+
+## Event Loop Explained 
+
+When Node.js starts, it initializes the event loop, processes the provided
+input script (or drops into the REPL, which is not covered in this document)
+which may make async API calls, schedule timers, or call `process.nextTick()`, 
+then begins processing the event loop.  
+
+The Following diagram shows a simplified overview of the event loop's
+order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+*note: each box will be referred to as a "phase" of the event loop.*
+
+Each phase has a FIFO queue of callbacks to execute. While each phase  is
+special in its own way, generally, when the event loop enters a  given phase,
+it will perform any operations specific to that phase,  then execute callbacks
+in that phase's queue until the queue has been exhausted or the maximum number
+of callbacks have executed.  When the queue has been exhausted or the callback
+limit is reached, the event loop will move to the next  phase, and so on.
+
+Since any of these operations may schedule _more_ operations and new events
+processed in the `poll` phase are queued by the kernel, poll events can be
+queued while polling events are being processed.  As a result, long running
+callbacks can allow the poll phase to run much longer than a timer's
+threshold.  See the [`timers`](#timers) and [`poll`](#poll) sections for more
+details.
+
+_**NOTE:** There is a slight discrepancy between the Windows and the
+Unix/Linux implementation, but that's not important for this demonstration.
+The most important parts are here.  There are actually seven or eight steps,
+but the ones we care about — ones that Node.js actually uses are these four._
+
+
+## Phases Overview:
+
+* `timers`: this phase executes callbacks scheduled by `setTimeout()` 
+  and `setInterval()`.
+* `pending callbacks`: this phase executes callbacks for specific types of TCP
+  errors, for example.
+* `poll`: this is the phase in which the event loop either processes
+  its queue of callbacks, or sits and waits for new callbacks for incoming
```

Comment:
poll phase doesn't do anything initially on its own. (cc @bnoordhuis) It will sit and wait to be notified on a file descriptor. At which time libuv will call the corresponding callback.